jbpm-flow/src/main/java/org/jbpm/process/instance/ProcessRuntimeImpl.java

```diff
@@ -436,10 +436,16 @@ public void matchCreated(MatchCreatedEvent event) {
                         kruntime.queueWorkingMemoryAction(new SignalManagerSignalAction(eventType, event));
                     } else if (ruleName.startsWith( "RuleFlowStateEventSubProcess-" ) 
                             || ruleName.startsWith( "RuleFlowStateEvent-" ) 
-                            || ruleName.startsWith( "RuleFlow-Milestone-" ) 
                             || ruleName.startsWith( "RuleFlow-AdHocComplete-" )
                             || ruleName.startsWith( "RuleFlow-AdHocActivate-" )) {
                         kruntime.queueWorkingMemoryAction(new SignalManagerSignalAction(ruleName, event));
+                    } else if (ruleName.startsWith("RuleFlow-Milestone-")) {
+                        String caseId = (String) kruntime.getEnvironment().get(EnvironmentName.CASE_ID);
+                        if (caseId != null) {
```

Comment:
When can it happen that case id is null?

Reply:
when you run with different runtime strategy than PER_CASE

Reply:
Ah, so when you use milestone in a process, not case?

Reply:
Correct as this is then expected behavior as we have no other context to bind to